### ISSUE TITLE
WING Firmware v3.0 causes an error

### ISSUE THREAD

**Issue by sackboydad** [bitfocus/companion-module-behringer-wing#24]
Looks like Behringer changed something in the communication protocol with WING firmware 3.0, which supports the three new mixers they announced yesterday.   

After loading FW 3.0, the yellow triangle appears in any button that calls on the WING module.  On the modules page, it shows an error "Cannot Get Status" (or something like that... sorry I had to leave and forgot to write the exact error down)

After downgrading to FW 2.1, the error went away.

**Comment by elliotmatson** [comment]
It looks like this is similar to #22. I'm currently in the middle of re-writing the module to support Companion v3.0. Once I'm done with that, I'll start adding some of the new requested features and support for Wing FW3.0

**Comment by sackboydad** [comment]
> It looks like this is similar to [#22](https://github.com/bitfocus/companion-module-behringer-wing/issues/22). I'm currently in the middle of re-writing the module to support Companion v3.0. Once I'm done with that, I'll start adding some of the new requested features and support for Wing FW3.0

Thank you.  Yeah, I had the yellow triangle warning in the "connections" tab in Companion since we bought the WiNG lsat February.  But never had the Yellow Triangle on any WING button itself until just today, after upgrading to FW 3.0.

**Comment by sackboydad** [comment]
FWIW, I re-installed FW3.0 and of course, the error triangle in all the WING buttons came back.  But I tested to see if the stuff actually worked.  Everything actually worked, including buttons updating with fader levels, fader commands, etc.   I still downgraded back to FW2.1 but thought I'd share a little testing with you.    Question, on your re-write do you plan on adding support for the WING-Live card?    Right now, I use generic OSC commands to control it but feedbacks or like position, etc would be nice.   Thanks for all you do on this module!

**Comment by elliotmatson** [comment]
Yes I do! I currently do the exact same thing with our Live card!

**Comment by sackboydad** [comment]
> Yes I do! I currently do the exact same thing with our Live card!

Hey, maybe in later FW, they've increased the subscription method to have more than one....

**Comment by istnv** [comment]
@andrewinalaska 
> can't seem to get companion to load edited modules from my dev path. I'm afraid I'm pretty wet behind the ears in this context. 

You would first need to recode the module for the companion 3.0 api before it will run from the dev directory.

> If I recall, I was working a change to fix how Fader Adjust works, so that it could work in dB instead of %-of-fader-travel... and a few other odds and ends. Anyway, if there's anything I can do to help as you dig into it, please let me know!

Fader 'set' works that way. Internally the console 'thinks' in percent and calculates dB on the fly. You would need to reverse the 'dB' to '%' calculation to send the proper value to the console.

**Comment by sackboydad** [comment]
> It looks like this is similar to [#22](https://github.com/bitfocus/companion-module-behringer-wing/issues/22). I'm currently in the middle of re-writing the module to support Companion v3.0. Once I'm done with that, I'll start adding some of the new requested features and support for Wing FW3.0

Hi Elliot…..  Are you at the beta stage yet?  I’d love to try it out.  I

**Comment by sackboydad** [comment]
Elliot.  Happy New Year!

I've been using the 3.50 beta with your new WING 2.x code.  I really like the way you've simplified the UI.  Really cleaned up over the first version.

I've been doing a lot of testing using the 3.5 beta with WING 3.05.   Obviously, it won't work properly with Matrices that use v3.x feature that makes them essentially mix busses.  Of the stuff we use at our church, I haven't found anything that works in the v1 or v2 of the WING companion module that doesn't work under WING FW3.05.   So, technically, I can easily install FW3.05 and use the current beta with your new UI and code.

The issue is the constant error than Companion gives when connected to WING FW3.x.  There's obviously some kind of response companion (or the module) wants to see that Behringer changed from firmware 1/2 to firmware 3.  Is there any way to have a switch in your module to ignore the warning and if there is, would you be willing to entertain adding that before release?   I've seen other modules have switches in the module setup turning on or off certain features.

I would like to take advantage of some of the new FW3 features, but it's really distracting to have every WING button on companion show the error symbol in the upper-right.

Thanks for your work an consideration on this request!

**Comment by elliotmatson** [comment]
Hello! Sorry, I'm just getting back into my routines after the holidays!

Just to clarify, I'm not a member of the Bitfocus team, so I didn't have anything to do with the new UI changes, all credit to them!

If you check the latest Pull Requests, Janik Witzig has started a full re-write of the module to make it easier to maintain, and I'm about to start testing that. 

I'll take a look and see if I can figure out the root cause of connection issue, but for some reason I'm not able to consistently get that error. Is there anything about your network that could be causing it?

**Comment by sackboydad** [comment]
Hi Elliot.  Thanks for writing back.  I totally get it.  Lots of other stuff going on.    My network can't be causing it.  If I'm on v2.1 of WING, there are no errors (other than the Companion warning that it hasn't been updated in a long time and might not be compatible).   When I go to v3, it just puts out an error "Warning Load Status".  If I go back again to v2.1 it goes away.

I'm guessing (and it's definitely a guess) that there's some acknologment or handshake that's been changed and it's different than the module is expecting.  With the obvious exception of new feature and actual changes they made to the WING in v3, all features that I've been using continue to work perfectly.  It's just the yellow triangle that is in every single stream deck button that does WING stuff that's annoying.

Thanks!

**Comment by janikwitzig** [comment]
Being worked on in #26
